src/sign.c

```diff
@@ -329,10 +325,14 @@ int wally_ec_sig_from_bytes(const unsigned char *priv_key, size_t priv_key_len,
         return WALLY_ENOMEM;
 
     if (flags & EC_FLAG_SCHNORR) {
+        secp256k1_keypair keypair;
         if (flags & EC_FLAG_RECOVERABLE)
             return WALLY_EINVAL; /* Only ECDSA is supported for recoverable sigs */
-        /* FIXME: Implement taproot schnorr sigs */
-        return WALLY_ERROR;
+        if (secp256k1_keypair_create(ctx, &keypair, priv_key) != 1)
+            return WALLY_ERROR;
+        if (secp256k1_schnorrsig_sign32(ctx, bytes_out, bytes, &keypair, /* aux_rand32 */ NULL) != 1)
+            return WALLY_ERROR;
```

Comment:
`keypair` needs to be cleared for both success and failure cases before returning here.

Reply:
done, I think